app/models/skin.rb

```diff
@@ -188,6 +188,7 @@ def cache!
     self.public = true
     self.official = true
     save!
+    Rails.cache.increment('skins_generation')
     css_to_cache = ""
```

Comment:
yes, I get the key-based expiration process just fine, what I don't understand is why you're doing it this way. Why are ALL skin caches being invalidated whenever ANY skin is cached? Wouldn't it make more sense to have something like "Rails.cache.increment("skins_generation_#{self.id}")"? 

Also, though, the cache!/clear_cache! methods in skins don't actually change the content of a skin at all? They just save the skin out to stylesheet files (or delete the stylesheet files), so I'm not really sure why you would want the cache to expire in these methods, and if you did, the skin model is touched in this step (because self.cached is set to true/false and it is saved). So the standard rails cache_key for the skin would be expired anyway?

I'm sure I'm just missing something here, but I don't want to stamp it until I understand--will try to catch you in chat to figure it out.